### ISSUE TITLE
Is the validation step computing gradients?

### ISSUE THREAD

**Issue by JuanseHevia** [wdoppenberg/ellipse-rcnn#13]
I noticed that you call set the Lightning Module into training mode during validation, is this computing gradients across the validation batch as well?

https://github.com/wdoppenberg/ellipse-rcnn/blob/72096ec44d1c8dfe23e6befa4fac0c19fd81cdf6/ellipse_rcnn/pl/model.py#L51

I tried commenting out the `self.train(True)` line, but then the model output is the actual predictions.

Maybe I'm not understanding the call correctly, could you clarify?

**Comment by wdoppenberg** [comment]
Hi! Yes this is indeed a bit strange. It is purely because you want to have the loss values, which are only returned (even in submodules) if `self.training == True`. This was inherited from `GeneralizedRCNN` and other native `torchvision` objects. 

I don't necessarily like it, so any suggestion to remove it is welcome. Also, I realized I didn't wrap the forward pass in the validation step with `torch.no_grad()`, so you might be able to use that in the short term.

**Comment by JuanseHevia** [comment]
Great, yes, so far I've been playing with the `LighitningModule.optimizers().zero_grad()` to ensure I'm not passing down gradients, but the `torch.no_grad()` seems a safer approach. Thanks.

I'll let you know if I come up with another approach. 

Great work btw!
